lib/dust.js

```diff
@@ -872,15 +873,21 @@
 
   Chunk.prototype.helper = function(name, context, bodies, params) {
     var chunk = this,
-        ret;
+        ret, filters;
     // handle invalid helpers, similar to invalid filters
     if(dust.helpers[name]) {
       try {
         ret = dust.helpers[name](chunk, context, bodies, params);
         if (dust.isThenable(ret)) {
-          return this.await(ret, context, bodies);
+          return chunk.await(ret, context, bodies);
+        }
+        if (ret instanceof Chunk) {
+          return ret;
+        }
+        if(typeof params.filters === 'string') {
+          filters = params.filters.split('|');
         }
-        return ret;
+        return chunk.reference(ret, context, 'h', filters);
```

Comment:
I know this isn't going to be easy but we should support `pragmas` and not hardcode the `h` here.

e.g.
```
{%esc:s}
  {@unsafe}{~n}
  {%esc:h}
    {@unsafe}
  {/esc}
{/esc}
```